stdlib/public/core/ObjectIdentifier.swift

```diff
@@ -14,6 +14,9 @@
 ///
 /// In Swift, only class instances and metatypes have unique identities. There
 /// is no notion of identity for structs, enums, functions, or tuples.
+///
+/// `ObjectIdentifier` is only guaranteed to remain unique for the
+/// lifetime of an object. 
```

Comment:
[Sorry if this seems like nit-picking but:] This doesn't sound quite right. `ObjectIdentifier` is a type. It is not the _type_ which is unique, values of that type are unique at runtime. This is why I suggested:

```
The unique identity for a class instance is only valid for comparisons during
the lifetime of the instance. 
```

The "unique identity" corresponds to values of type `ObjectIdentifier`, and IMO fits better stylistically with the rest of the doc comment, which also uses "unique identities" as shorthand for "values of type `ObjectIdentifier`".